src/OpenColorIO/ops/Lut3D/Lut3DOpData.cpp

```diff
@@ -0,0 +1,1048 @@
+/*
+Copyright (c) 2018 Autodesk Inc., et al.
+All Rights Reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+* Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+* Neither the name of Sony Pictures Imageworks nor the names of its
+  contributors may be used to endorse or promote products derived from
+  this software without specific prior written permission.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include <sstream>
+
+#include <OpenColorIO/OpenColorIO.h>
+
+#include "BitDepthUtils.h"
+#include "HashUtils.h"
+#include "MathUtils.h"
+#include "md5/md5.h"
+#include "ops/Lut3D/Lut3DOp.h"
+#include "ops/Lut3D/Lut3DOpData.h"
+#include "ops/Range/RangeOpData.h"
+#include "OpTools.h"
+#include "Platform.h"
+
+OCIO_NAMESPACE_ENTER
+{
+
+// Allow us to temporarily manipulate the style without cloning the object.
+class Lut3DStyleGuard
+{
+public:
+    Lut3DStyleGuard(const Lut3DOpDataRcPtr & lut)
+        : m_lut(lut)
+        , m_wasFast(lut->getInvStyle() == Lut3DOpData::INV_FAST)
+    {
+        m_lut->setInvStyle(Lut3DOpData::INV_EXACT);
+    }
+
+    ~Lut3DStyleGuard()
+    {
+        if (m_wasFast)
+        {
+            m_lut->setInvStyle(Lut3DOpData::INV_FAST);
+        }
+    }
+
+private:
+    Lut3DOpDataRcPtr m_lut;
+    bool             m_wasFast;
+};
+
+Lut3DOpDataRcPtr MakeFastLut3DFromInverse(const Lut3DOpDataRcPtr & lut)
```

Comment:
If this function is using the EXACT style always, should it be renamed to `MakeExactLut3DFromInverse`? Or is there a use case to simply honor the style set by the `Lut3DOpDataRcPtr`? I see this is used for inversion in the GPU renderer, and this would imply that only the CPU renderer can use the FAST inversion style, thus making a difference in CPU/GPU calculations if that style were requested.

Reply:
Ok, I see what you're saying. I was misreading the FAST implementation overall. Thanks for the clarification.